packages/material-ui/src/Badge/Badge.js

```diff
@@ -186,6 +186,10 @@ const Badge = React.forwardRef(function Badge(props, ref) {
     displayValue = badgeContent > max ? `${max}+` : badgeContent;
   }
 
+  if (invisible && badgeContent === 0 && variant !== 'dot') {
+    displayValue = '1';
```

Comment:
It depends on the previous `badgeContent` value, so it can be `2` or more (we also use `ON`).

Reply:
If `useState()` is overkill I'll close this PR. :bow:

Reply:
Ah, if we need to to solve `badgeContent === 0` only, we can `if (badgeContent === 0) displayValue = ''`.

Reply:
Yeah, it's not perfect, I only handle the case where the value jumps from 0 to 1. OK, I think that we can give up on the idea. We don't have time to look at details of this level. I think that it would be better to leave it to userland until we have enough time to deticate to it. The initial ref approach only hanldes the first render. It's incomplete.

Could you revert the change in the Badge so we can merge the docs changes? :).